  * Error out with an explicit message when trying to initialize a
    character array with something that's not a literal (optionally
    enclosed in braces) as per C99 6.7.8:14; thanks to Antti-Juhani
    Kaijanaho <ajk@debian.org> who did all the work (closes: #259619).
Index: tcc-0.9.24~cvs20070502.orig/tcc.c
===================================================================
--- tcc-0.9.24~cvs20070502.orig.orig/tcc.c	2007-05-03 03:37:24.934358000 +0200
+++ tcc-0.9.24~cvs20070502.orig/tcc.c	2007-05-03 03:55:29.955523709 +0200
@@ -8737,6 +8737,9 @@
         no_oblock = 1;
         if ((first && tok != TOK_LSTR && tok != TOK_STR) || 
             tok == '{') {
+            if (tok != '{')
+                error("character array initializer must be a literal,"
+                    " optionally enclosed in braces");
             skip('{');
             no_oblock = 0;
         }
